ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -632,25 +637,52 @@ export default class ConfirmTransactionBase extends Component {
   }
 
   renderData(functionType) {
+    const { t } = this.context;
+    const { txData: { txParams } = {}, hideData, dataComponent } = this.props;
+
+    if (hideData) {
+      return null;
+    }
+
+    return (
+      dataComponent || (
+        <div className="confirm-page-container-content__data">
+          <div className="confirm-page-container-content__data-box-label">
+            {`${t('functionType')}:`}
+            <span className="confirm-page-container-content__function-type">
+              {functionType}
+            </span>
+          </div>
+          <Disclosure>
+            <TransactionDecoding to={txParams?.to} inputData={txParams?.data} />
+          </Disclosure>
+        </div>
+      )
+    );
+  }
+
+  renderDataHex(functionType) {
     const { t } = this.context;
     const {
-      txData: { txParams: { data } = {} } = {},
+      txData: { txParams } = {},
       methodData: { params } = {},
       hideData,
-      dataComponent,
+      dataHexComponent,
     } = this.props;
 
     if (hideData) {
       return null;
     }
 
     return (
-      dataComponent || (
+      dataHexComponent || (
         <div className="confirm-page-container-content__data">
           <div className="confirm-page-container-content__data-box-label">
             {`${t('functionType')}:`}
             <span className="confirm-page-container-content__function-type">
-              {functionType}
+              {`${functionType} (${
+                params ? params.map(({ type }) => type).join(', ') : null
```

Comment:
`null` will be rendered as a string here, this should be an empty string instead